agent/pkg/api/socket_routes.go

```diff
@@ -94,72 +95,84 @@ func websocketHandler(w http.ResponseWriter, r *http.Request, eventHandlers Even
 	startTimeBytes, _ := models.CreateWebsocketStartTimeMessage(startTime)
 	SendToSocket(socketId, startTimeBytes)
 
+	queryRecieved := false
+
 	for {
 		_, msg, err := ws.ReadMessage()
 		if err != nil {
 			logger.Log.Errorf("Error reading message, socket id: %d, error: %v", socketId, err)
 			break
 		}
 
-		if !isTapper && !isQuerySet {
-			query := string(msg)
-			err = basenine.Validate(shared.BasenineHost, shared.BaseninePort, query)
-			if err != nil {
-				toastBytes, _ := models.CreateWebsocketToastMessage(&models.ToastMessage{
-					Type:      "error",
-					AutoClose: 5000,
-					Text:      fmt.Sprintf("Syntax error: %s", err.Error()),
-				})
-				SendToSocket(socketId, toastBytes)
-				break
-			}
+		if !queryRecieved {
```

Comment:
@mertyildiran 
I'm probably miss something - and I'm not sure I understand the "queryRecieved" here.
can you help me understand what is the idea here?

Reply:
If I understand correctly, this for loop is happening when opening connection (this happen every time filter is applied), and the loop will start send information to the browser via the web socket connection.
so this change allowing the client (UI) to send message to the server when entry is changed and (
```

        if (ws.current.readyState === WebSocket.OPEN) {
            ws.current.send(focusedEntryId);
        }
```

and if I understand correctly the server will just return message with the same data?

Reply:
@gadotroee kind of yes.

The first ever message from client-to-server is the query like `request.path == "..."` The subsequent messages can only be the `focusedEntryId` values and yes we simply mirror the value back to the client. It's required to resolve the racing issues (finally I understand this time). So;

This from this line:

https://github.com/up9inc/mizu/blob/a62fba4cb63fdb29873633f765115c1ff777b014/ui/src/components/TrafficPage.tsx#L205-L207

value is passed to this block:

https://github.com/up9inc/mizu/blob/a62fba4cb63fdb29873633f765115c1ff777b014/ui/src/components/TrafficPage.tsx#L170-L177

through WebSocket. To prevent race condition that's caused by this block:

https://github.com/up9inc/mizu/blob/a62fba4cb63fdb29873633f765115c1ff777b014/ui/src/components/TrafficPage.tsx#L127-L138

Reply:
@mertyildiran 
Ok, I understand it now and it should work - I Just think if it is good practice to assume that all messages after query passed are only messages of focusedEntryId (it might be Ok because before we didn't use the client to server connection after making the query and it was used only for server to client messaged (different types including data / metadata (tapped pods) / maybe another type that  I forget about).

Also I think it will be different after the pagination change.. 
so I will approve this to get "working" develop.